c/src/core/codec.c

```diff
@@ -431,6 +431,7 @@ void pn_data_clear(pn_data_t *data)
     data->current = 0;
     data->base_parent = 0;
     data->base_current = 0;
+    data->error = NULL;
```

Comment:
Pretty sure this code is wrong - despite passing the tests:
If there was a an error attached to the pn_data_t then it would be leaked by just setting the field to NULL. This needs ```pn_error_free(data->error);``` first.
As an alternative you could use ```pn_error_clear(data->error);``` instead, which will overall do a little less work whilst giving the same effect.

Reply:
On second thought, I have no idea why I just set that to `NULL`. Thanks for providing me the correct utility function.
Changed the PR to use `pn_error_clear(data->error);` now.